nullaway/src/main/java/com/uber/nullaway/handlers/contract/fieldcontract/EnsuresNonNullHandler.java

```diff
@@ -153,14 +164,18 @@ public NullnessHint onDataflowVisitMethodInvocation(
       fieldNames = ContractUtils.trimReceivers(fieldNames);
       for (String fieldName : fieldNames) {
         VariableElement field =
-            getInstanceFieldOfClass(
-                castToNonNull(ASTHelpers.enclosingClass(methodSymbol)), fieldName);
+            getFieldOfClass(castToNonNull(ASTHelpers.enclosingClass(methodSymbol)), fieldName);
         if (field == null) {
           // Invalid annotation, will result in an error during validation. For now, skip field.
           continue;
         }
-        AccessPath accessPath =
-            AccessPath.fromBaseAndElement(node.getTarget().getReceiver(), field, apContext);
+        AccessPath accessPath;
+        if (field.getModifiers().contains(Modifier.STATIC)) {
+          accessPath = AccessPath.fromStaticField(field);
+        } else {
+          accessPath =
+              AccessPath.fromBaseAndElement(node.getTarget().getReceiver(), field, apContext);
+        }
```

Comment:
Might be simpler to just use a conditional expression, `AccessPath accessPath = field.getModifiers().contains(Modifier.STATIC) ? ... : ...;`

Reply:
Fixed!